src/main/java/in/org/projecteka/hdaf/clients/UserServiceClient.java

```diff
@@ -0,0 +1,29 @@
+package in.org.projecteka.hdaf.clients;
+
+import in.org.projecteka.hdaf.clients.properties.UserServiceProperties;
+import in.org.projecteka.hdaf.link.discovery.model.User;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+public class UserServiceClient {
+
+    private final WebClient.Builder webClientBuilder;
+    private final UserServiceProperties userServiceProperties;
+
+    public UserServiceClient(
+            WebClient.Builder webClientBuilder,
+            UserServiceProperties userServiceProperties) {
+        this.webClientBuilder = webClientBuilder;
+        this.userServiceProperties = userServiceProperties;
+    }
+
+    public Mono<User> userOf(String userId) {
+        return webClientBuilder.build()
+                .get()
+                .uri(String.format("%s/users/%s/", userServiceProperties.getUrl(), userId))
+                .header("X-Auth-Token", userServiceProperties.getXAuthToken())
```

Comment:
It can't be like this, need to have a auth client, which will give us auth-token for internal service communication. If it's not required remove it for now.